File: scripts/download-images.js
import fs from 'fs';
import path from 'path';
import https from 'https';

// 使用绝对路径
const imagesDir = '/vercel/share/v0-project/public/images';

console.log('[v0] 图片目录:', imagesDir);

// 确保目录存在
try {
  if (!fs.existsSync(imagesDir)) {
    fs.mkdirSync(imagesDir, { recursive: true });
  }
} catch (error) {
  console.log('[v0] 目录已存在或权限受限，继续下载...');
}

const images = [
  {
    url: 'https://hebbkx1anhila5yf.public.blob.vercel-storage.com/Frame%207132-HJpuPiROV0R54MEiuyndeK4CqVAKqh.png',
    filename: 'solution-middleware.png',
    name: '信创中间件双活容灾'
  },
  {
    url: 'https://hebbkx1anhila5yf.public.blob.vercel-storage.com/Subtract%20%281%29-vsE5hWqJd0UpujSCdARKbC223ITKh2.png',
    filename: 'solution-xinchuang.png',
    name: '信创全栈自主可控'
  },
  {
    url: 'https://hebbkx1anhila5yf.public.blob.vercel-storage.com/Frame%207131%20%281%29-BPlmefIpxt6it2acm5teAnn8OjfBKf.png',
    filename: 'solution-ai-agent.png',
    name: 'AI智能体管理平台'
  }
];

function downloadImage(url, filepath) {
  return new Promise((resolve, reject) => {
    const file = fs.createWriteStream(filepath);
    https.get(url, (response) => {
      if (response.statusCode !== 200) {
        reject(new Error(`Failed to download: ${response.statusCode}`));
        return;
      }
      response.pipe(file);
      file.on('finish', () => {
        file.close();
        resolve();
      });
    }).on('error', reject);
  });
}

async function downloadAllImages() {
  console.log('[v0] 开始下载解决方案图片...\n');
  
  for (const image of images) {
    const filepath = path.join(imagesDir, image.filename);
    try {
      console.log(`[v0] 下载: ${image.name}`);
      await downloadImage(image.url, filepath);
      const stats = fs.statSync(filepath);
      console.log(`[v0] ✓ 已保存: ${image.filename} (${(stats.size / 1024 / 1024).toFixed(2)} MB)\n`);
    } catch (error) {
      console.error(`[v0] ✗ 下载失败 ${image.filename}: ${error.message}\n`);
    }
  }
  
  console.log('[v0] 下载完成！');
  console.log('[v0] 这些图片现在已保存到项目中，可以离线查看。\n');
}

downloadAllImages().catch(console.error);

File: scripts/download-images.py
#!/usr/bin/env python3
"""
Download and optimize critical images for local deployment
"""

import os
import requests
from pathlib import Path
from PIL import Image
from io import BytesIO

# Create images directory
BASE_DIR = Path(".")
IMAGES_DIR = BASE_DIR / "public" / "images"
IMAGES_DIR.mkdir(parents=True, exist_ok=True)

# Critical images mapping
IMAGES = {
    # Banner images
    "https://hebbkx1anhila5yf.public.blob.vercel-storage.com/Group%20111%20%286%29-XkEWWQ1R1ZVnSsnrH13QZ3KZJpP4xX.png": "home/banner-careers.jpg",
    "https://images.unsplash.com/photo-1552664730-d307ca884978?w=1920&q=80&auto=format": "home/customers.jpg",
    "https://images.unsplash.com/photo-1552664730-d307ca884978?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&q=80&w=1920": "home/honors.jpg",
    
    # Footer QR code
    "https://hebbkx1anhila5yf.public.blob.vercel-storage.com/qrcode-2DUnWMBfNJhVmKGN6YqnvC8K5Wbp8B.png": "footer/qrcode.png",
    
    # Online service icon  
    "https://hebbkx1anhila5yf.public.blob.vercel-storage.com/%E5%9C%A8%E7%BA%BF-9UdxG92GVQ7UZ44cfJw5DbpK3kU5VT.png": "icons/online-service.png",
    
    # Solution diagrams
    "https://hebbkx1anhila5yf.public.blob.vercel-storage.com/%E6%88%AA%E5%B1%8F2026-02-26%2009.48.54-dINeUw8HwaEHihoN4MlRtqH4ruXSaP.png": "diagrams/middleware-arch.png",
    "https://hebbkx1anhila5yf.public.blob.vercel-storage.com/%E6%9E%B6%E6%9E%84%E5%9B%BE-qte6VDj3U4iS74yOekcWo9DCTojP0r.png": "diagrams/ai-agent-arch.png",
}

print("Starting image download and optimization...")
downloaded_count = 0
failed_count = 0

for url, local_path in IMAGES.items():
    try:
        print(f"Processing: {local_path}...")
        
        # Create subdirectory
        local_file = IMAGES_DIR / local_path
        local_file.parent.mkdir(parents=True, exist_ok=True)
        
        # Download image
        response = requests.get(url, timeout=30)
        response.raise_for_status()
        
        # Open and optimize image
        img = Image.open(BytesIO(response.content))
        
        # Convert to RGB if necessary (remove alpha channel)
        if img.mode in ('RGBA', 'LA', 'P'):
            rgb_img = Image.new('RGB', img.size, (255, 255, 255))
            rgb_img.paste(img, mask=img.split()[-1] if img.mode == 'RGBA' else None)
            img = rgb_img
        
        # Resize if too large
        max_width = 1920 if 'banner' in local_path or 'arch' in local_path else 800
        if img.width > max_width:
            ratio = max_width / img.width
            new_height = int(img.height * ratio)
            img = img.resize((max_width, new_height), Image.Resampling.LANCZOS)
        
        # Save optimized
        quality = 80 if 'banner' in local_path or 'arch' in local_path else 85
        img.save(str(local_file), 'JPEG', quality=quality, optimize=True)
        
        print(f"✓ Downloaded and optimized: {local_path}")
        downloaded_count += 1
        
    except Exception as e:
        print(f"✗ Failed to download {local_path}: {str(e)}")
        failed_count += 1

print(f"\n完成: {downloaded_count} 个成功, {failed_count} 个失败")
